app/uk/gov/hmrc/hec/models/HECTaxCheckData.scala

```diff
@@ -16,16 +16,27 @@
 
 package uk.gov.hmrc.hec.models
 
-import julienrf.json.derived
-import play.api.libs.json.OFormat
+import ai.x.play.json.Jsonx
+import ai.x.play.json.SingletonEncoder.simpleName
+import ai.x.play.json.implicits.formatSingleton
+import play.api.libs.json.{Format, JsObject, JsResult, JsValue, Json, OFormat}
 import uk.gov.hmrc.hec.models.ApplicantDetails.{CompanyApplicantDetails, IndividualApplicantDetails}
 import uk.gov.hmrc.hec.models.TaxDetails.{CompanyTaxDetails, IndividualTaxDetails}
 import uk.gov.hmrc.hec.models.licence.LicenceDetails
 
-sealed trait HECTaxCheckData extends Product with Serializable {
+sealed trait EntityType extends Product with Serializable
```

Comment:
just in case we use it in other places in the future and for general neatness it might make sense to move `EntityType` out in to it's own file

Reply:
updated in latest commits